mlflow/store/tracking/abstract_store.py

```diff
@@ -940,6 +941,39 @@ def link_traces_to_run(self, trace_ids: list[str], run_id: str) -> None:
             MlflowException: If more than 100 traces are provided.
         """
 
+    @abstractmethod
+    def calculate_trace_filter_correlation(
+        self,
+        experiment_ids: list[str],
+        filter_string1: str,
+        filter_string2: str,
+    ) -> TraceFilterCorrelationResult:
+        """
+        Calculate correlation between two trace filter conditions using NPMI.
+
+        This method analyzes the correlation between traces matching two different
+        filter conditions using Normalized Pointwise Mutual Information (NPMI).
+
+        Args:
+            experiment_ids: List of experiment IDs to analyze traces from.
+            filter_string1: First filter condition (MLflow search filter syntax).
+            filter_string2: Second filter condition (MLflow search filter syntax).
+
+        Returns:
+            TraceFilterCorrelationResult containing:
+            - npmi: Correlation score from -1 (never co-occur) to 1 (always co-occur),
+                   or NaN if undefined (when a filter has zero matches)
+            - filter1_count: Number of traces matching filter1
+            - filter2_count: Number of traces matching filter2
+            - joint_count: Number of traces matching both filters
+            - total_count: Total number of traces in the experiments
+            - expected_joint: Expected joint count under independence
+            - lift: Ratio of observed to expected joint count
+
+        Raises:
+            MlflowException: If filters are invalid or experiments don't exist.
+        """
```

Comment:
Can we raise `NotImplemented` by default? Else, I think that existing stores that inherit from `AbstractStore` may fail to instantiate because they don't override this abstract method